packages/babel-plugin-transform-classes/test/fixtures/spec/method-check-new-extends/expected.js

```diff
@@ -0,0 +1,23 @@
+var Foo =
+/*#__PURE__*/
+function (_Bar) {
+  babelHelpers.inherits(Foo, _Bar);
+
+  function Foo() {
+    var _this;
+
+    babelHelpers.classCallCheck(this, Foo);
+    _this = babelHelpers.possibleConstructorReturn(this, (Foo.__proto__ || Object.getPrototypeOf(Foo)).call(this));
+    _this.state = "test";
+    return _this;
+  }
+
+  babelHelpers.createClass(Foo, [{
+    key: "bar",
+    value: function bar() {
+      babelHelpers.newMethodCheck(this, Foo.prototype.bar);
```

Comment:
It seems like it'd be possible for this to reference `bar` directly. Is there any reason to reference it from the prototype like this?